src/main/java/com/firebolt/jdbc/client/query/StatementClientImpl.java

```diff
@@ -284,6 +285,19 @@ private Map<String, String> getCancelParameters(String statementId) {
 		return Map.of(FireboltQueryParameterKey.QUERY_ID.getKey(), statementId);
 	}
 
+
+	@Override
+	protected void validateResponse(String host, Response response, Boolean isCompress) throws FireboltException {
+		super.validateResponse(host, response, isCompress);
+		FireboltConnection connection = getConnection();
+		if (isCallSuccessful(response.code())) {
+			for (String header : response.headers(HEADER_UPDATE_PARAMETER)) {
+				String[] keyValue = header.split("=");
+				connection.addProperty(Pair.of(keyValue[0].trim(), keyValue[1].trim()));
```

Comment:
I think we should only add known parameters from this header (as we do in python).
Just have a list of parameters that we know (basically only "database") and add a property only in case key is in this list

Reply:
Definitely not. This will work for `set` statement too.

Reply:
We don't yet have `set` statements and we don't know yet how they will be implemented. The header approach had some issues so it was postponed. I believe we should enable this functionality excplicitly to make sure it's tested before it's exposed to customers and not just leave backdoors for it to work in the future